### ISSUE TITLE
Haste RNN support

### ISSUE THREAD

**Issue by thegodone** [aimat-lab/gcnn_keras#8]
Do you see a way to replace the keras GRU by the Haste_tf GRU version from here https://github.com/lmnt-com/haste/blob/master/docs/tf/haste_tf.md ?

there are lot of advantage in this package for regularization at cuda level etc...

unfortunately I don't see a way to plug the haste_tf at keras level https://github.com/lmnt-com/haste/issues/33 ?

your help would be appreciate

**Comment by thegodone** [comment]
I was able to do this : 

import numpy as np
import tensorflow as tf
import haste_tf as haste


train_x =  np.random.rand(500,40,20)
train_y =  np.random.rand(500,40,1)


# RMSE definition
def root_mean_squared_error(y_true, y_pred):
        return K.sqrt(K.mean(K.square(y_pred - y_true))) 

inputs = tf.keras.Input(shape=(train_x.shape[1], train_x.shape[2]))
haste1 = haste.LayerNormGRU(20, direction='unidirectional', zoneout=0.1, dropout=0.1)
fc1 = tf.keras.layers.Dense(60, activation='relu', kernel_initializer='he_uniform')
dr1 = tf.keras.layers.Dropout(0.2)
fc2 = tf.keras.layers.Dense(1)

x, state = haste1(inputs, training=True)
x = fc1(inputs)
x = dr1(x)
outputs = fc2(x)

model = tf.keras.Model(inputs=inputs, outputs=outputs)

print(model.summary())

opt = tf.keras.optimizers.Adam(learning_rate=0.01)
model.compile(loss='categorical_crossentropy', optimizer=opt)

#model.compile(loss=root_mean_squared_error, optimizer=optimizer)
model_hist = model.fit(train_x, train_y, epochs=10, batch_size=32, verbose=1)

is it possible to add this layer in place of pure "GRU" ?

**Comment by PatReis** [comment]
I will look into it. However, since kgcnn is fully based on keras, I am not fully sure if we sould add more external dependencies apart from keras. But maybe as a layer module is should be fine.

**Comment by thegodone** [comment]
I really want to test this special layer if it's possible. Maybe you can an exception for this one.

**Comment by thegodone** [comment]
there is a typo here
![image](https://user-images.githubusercontent.com/1186658/123904222-6a6c9e80-d970-11eb-879e-d51ca3ee59a3.png)

"dopout" => "dropout"

**Comment by thegodone** [comment]
another issue than
![image](https://user-images.githubusercontent.com/1186658/123905028-d6033b80-d971-11eb-8b30-00accfc4dfb9.png)

**Comment by PatReis** [comment]
Thanks, I am still trying to get haste installed. But I think this error is not from us. We must post this error to haste, since I think there must be a 
`tf.compat.v1.get_variable("gamma",shape=1,initializer=tf.compat.v1.keras.initializers.ones())`
since
`tf.compat.v1.get_variable("gamma",initializer=tf.compat.v1.keras.initializers.ones())`
will always throw an error regardless of the input.

I am not sure that there is a great advantage using haste Cells. I think there is an advantage using the final haste GRU. But because the recurrent step requires some attention pooling over the nodes (for graph networks) we are stuck with the GRUCells. However, I will think about this, maybe I did make a mistake in my logic/implementation.

**Comment by thegodone** [comment]
you want to open a issue in Haste. Go ahead

Le mer. 30 juin 2021 à 10:59, Patrick Reiser ***@***.***> a
écrit :

**Comment by PatReis** [comment]
Okay, thanks, I opened and issue in haste.
